src/test/clojure/clara/test_rules.clj

```diff
@@ -3004,3 +3004,23 @@
 
       (is (= {:?w 10, :?t nil}
              (-> e (ex-data) :bindings))))))
+
+;; Validate that the inital vector will remain as a vector even after a retraction occurs
+;; See https://github.com/cerner/clara-rules/issues/338
+(deftest test-data-structure-consistency-post-retraction
+  (let [cold-fact (->Cold -32)
+        warmer-fact (->Cold -31)
+        cold-query (dsl/parse-query [] [[?all-cold <- (acc/all) :from [Cold]]])
+        get-cold-facts #(-> (query % cold-query)
+                            first
+                            :?all-cold)
+        session (-> (mk-session [cold-query])
+                    (insert cold-fact warmer-fact)
+                    fire-rules
+                    (retract cold-fact)
+                    fire-rules)
+        cold-facts (get-cold-facts session)]
+
+    (is (and (= [warmer-fact] cold-facts)
```

Comment:
This isn't a big deal since this isn't the focus of this test, but ideally we'd validate that there is only one query result as well IMO.  So

````
(= 1 (count (query session cold-query)))
````